compiler/rustc_mir/src/interpret/intrinsics.rs

```diff
@@ -554,4 +558,18 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
         self.memory.copy(src, align, dst, align, size, nonoverlapping)
     }
+
+    pub(crate) fn raw_eq_intrinsic(
+        &mut self,
+        lhs: &OpTy<'tcx, <M as Machine<'mir, 'tcx>>::PointerTag>,
+        rhs: &OpTy<'tcx, <M as Machine<'mir, 'tcx>>::PointerTag>,
+    ) -> InterpResult<'tcx, Scalar<M::PointerTag>> {
+        let layout = self.layout_of(lhs.layout.ty.builtin_deref(true).unwrap().ty)?;
```

Comment:
Please add an assertion for `!layout.is_unsized()`

Reply:
this was not addressed

Reply:
It's there: https://github.com/rust-lang/rust/pull/85828/files#diff-8c6d2638626cb722d5bf6029e18df61437e73d41e43e9ee524966dab35bb0f1dR568